ppoprf/src/ppoprf.rs

```diff
@@ -87,6 +135,115 @@ impl ProofDLEQ {
     c_prime == self.c
   }
 
+  fn verify_batch(
+    &self,
+    public_value: &RistrettoPoint,
+    p: &[RistrettoPoint], //P
+    q: &[RistrettoPoint], //Q
+  ) -> bool {
+    //(M, Z) = ComputeComposites(B, C, D)
+    let (m, z) = ProofDLEQ::compute_composites(None, public_value, p, q);
+
+    //t2 = ((s * A) + (c * B))
+    let t2 = (self.s * RISTRETTO_BASEPOINT_POINT) + (self.c * public_value);
+    //t3 = ((s * M) + (c * Z))
+    let t3 = (self.s * m) + (self.c * z);
+
+    /*challengeTranscript =
+    I2OSP(len(Bm), 2) || Bm ||
+    I2OSP(len(a0), 2) || a0 ||
+    I2OSP(len(a1), 2) || a1 ||
+    I2OSP(len(a2), 2) || a2 ||
+    I2OSP(len(a3), 2) || a3 ||
+    "Challenge"*/
+    let mut challenge_transcript = Vec::new();
+    let compressed_point_len_slice = &ProofDLEQ::i2osp2(COMPRESSED_POINT_LEN);
+    challenge_transcript.extend_from_slice(compressed_point_len_slice); //len(Bm)
+    challenge_transcript.extend_from_slice(public_value.compress().as_bytes()); //Bm
+    challenge_transcript.extend_from_slice(compressed_point_len_slice); //len(a0)
+    challenge_transcript.extend_from_slice(m.compress().as_bytes()); //a0 = m
+    challenge_transcript.extend_from_slice(compressed_point_len_slice); //len(a1)
+    challenge_transcript.extend_from_slice(z.compress().as_bytes()); //a1 = z
+    challenge_transcript.extend_from_slice(compressed_point_len_slice); //len(a2)
+    challenge_transcript.extend_from_slice(t2.compress().as_bytes()); //a2 = t2
+    challenge_transcript.extend_from_slice(compressed_point_len_slice); //len(a3)
+    challenge_transcript.extend_from_slice(t3.compress().as_bytes()); //a3 = t3
+
+    //c = G.HashToScalar(challengeTranscript)
+    let c = ProofDLEQ::hash_to_scalar(&challenge_transcript, "Challenge");
+
+    //verified = (expectedC == c)
+    self.c == c
+  }
+
+  fn compute_composites(
+    key: Option<RistrettoScalar>,
+    b: &RistrettoPoint,
+    c: &[RistrettoPoint],
+    d: &[RistrettoPoint],
+  ) -> (RistrettoPoint, RistrettoPoint) {
+    if c.len() != d.len() {
+      panic!("C and D have a different number of elements!");
+    }
+
+    // We use the Partially-punctureable Oblivious Pseudo-Random Function
+    // We assign mode 0x03 for the PPOPRF
+    let context_string =
+      format!("{}-{}-{}", "PPOPRFv1", 0x03, "ristretto255-strobe");
+    //seedDST = "Seed-" || contextString
+    let seed_dst = format!("{}-{}", "Seed", context_string);
+
+    // seedTranscript = I2OSP(len(Bm), 2) || Bm || I2OSP(len(seedDST), 2) || seedDST
+    let mut seed_transcript = Vec::new();
+    seed_transcript.extend_from_slice(&ProofDLEQ::i2osp2(COMPRESSED_POINT_LEN)); //len(Bm)
+    seed_transcript.extend_from_slice(b.compress().as_bytes()); //Bm
+    seed_transcript.extend_from_slice(&ProofDLEQ::i2osp2(seed_dst.len())); //len(seedDST)
+    seed_transcript.extend_from_slice(seed_dst.as_bytes()); //seedDST
+
+    let mut seed = [0u8; DIGEST_LEN];
+    strobe_hash(&seed_transcript, "Seed", &mut seed);
```

Comment:
I think on all of these hashes, it is easier to follow the approach already present in the code, like in here: https://github.com/brave/sta-rs/blob/main/ppoprf/src/ppoprf.rs#L337 Right now, you are passing twice the "Seed" string, which is not needed.